src/common/globals.js

```diff
@@ -265,24 +269,40 @@ define([
             valueType: 'section'
         });
 
-        const storageMetadata = Storage.getAvailableBackends().map(id => Storage.getStorageMetadata(id));
+        if(pluginName === IMPORT_PLUGIN){
+            backends = storageBackends.filter(id => id !== 'gme');
+        }
+
         metadata.configStructure.push({
             name: 'storage',
             displayName: 'Storage',
             description: 'Location to store intermediate/generated data.',
             valueType: 'dict',
-            value: Storage.getBackend(Storage.getAvailableBackends()[0]).name,
-            valueItems: storageMetadata,
+            value: Storage.getBackend(backends[0]).name,
+            valueItems: storageMetadata.filter(metadata => backends.includes(metadata.id)),
         });
+        return metadata;
+    };
 
+    const runArtifactPlugin = async function(pluginName) {
+        const metadata = getConfigMetadata(pluginName);
         const configDialog = new ConfigDialog(client);
         const allConfigs = await configDialog.show(metadata);
-        const context = client.getCurrentPluginContext(UPLOAD_PLUGIN);
-        context.pluginConfig = allConfigs[UPLOAD_PLUGIN];
+        const context = client.getCurrentPluginContext(pluginName);
+        context.pluginConfig = allConfigs[pluginName];
         context.pluginConfig.storage.id = storageMetadata
             .find(metadata => metadata.name === context.pluginConfig.storage.name)
             .id;
-        return await Q.ninvoke(client, 'runBrowserPlugin', UPLOAD_PLUGIN, context);
+        return await Q.ninvoke(client, 'runBrowserPlugin', pluginName, context);
+    };
+
+
+    DeepForge.create.Artifact = async function() {
+        await runArtifactPlugin(UPLOAD_PLUGIN);
+    };
+
+    DeepForge.create.ImportPaths = async function() {
```

Comment:
So the `DeepForge.create.TYPE` methods are expecting the function name to be the type to create. This is breaking this convention as `ImportPaths` is not a type.

It would be better to add a flag to the the `DeepForge.create.Artifact` function like `importExisting=false`. Still not a perfect but at least conforms to the convention for the `DeepForge` object.

Reply:
How about we define new `Deepforge.Import.Artifact` define the function to run `ImportArtifact` on the function?

Reply:
Deepforge.import.Artifact will now run the `IMPORT_PLUGIN`.